ls1intum/Artemis
# `General`: Add deletion of multiple exercises

### Checklist
#### General



#### Client


### Motivation and Context
This PR is about the exercise overview in the course management. It adds a selection (like the one we already had for programming exercises) for all exercise types and adds a "Delete Selected" Action for all exercise types.
Closes #7181 

### Description
The selection for the other exercise types has been adapted from the already existing selection for programming exercises (introduced `selectedExercises` object) and the delete Action is - after confirmation - called for all the selected exercises.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 exercises for each exercise type

1. Log in to Artemis
2. Navigate to Course Management and Exercise Overview
3. Verify that it is possible to select the exercises and select all works for all exercise types
4. Verify that the selected exercises are deleted once the "Delete" Button is clicked


### Review Progress






#### Code Review
#### Manual Tests


### Test Coverage







### Screenshots


![image](https://github.com/ls1intum/Artemis/assets/78978542/0d66a2d6-8412-47d9-a764-cf27927413f0)
